web/app/components/document/sidebar/related-resources/add.ts

```diff
@@ -0,0 +1,292 @@
+   */
+  @tracked protected queryIsURL = false;
+
+  /**
+   * The DOM element of the search input. Receives focus when inserted.
+   */
+  @tracked private searchInput: HTMLInputElement | null = null;
+
+  /**
+   * Whether to allow navigating with the keyboard.
+   * True unless the search input has lost focus.
+   */
+  @tracked keyboardNavIsEnabled = true;
+
+  /**
+   * The title of the external link, as set by the optional input.
+   * Used to set the name of the external resource.
+   */
+  @tracked externalLinkTitle = "";
+
+  /**
+   * Whether the URL already exists as a related resource.
+   * Used to prevent duplicates in the array.
+   */
+  @tracked linkIsDuplicate = false;
+
+  /**
+   * Whether a query has no results.
+   * May determine whether the list header (e.g., "suggestions," "results") is shown.
+   */
+  private get noMatchesFound(): boolean {
+    const objectEntriesLengthIsZero =
+      Object.entries(this.args.shownDocuments).length === 0;
+
+    if (this.args.allowAddingExternalLinks) {
+      return objectEntriesLengthIsZero && !this.queryIsURL;
+    } else {
+      return objectEntriesLengthIsZero;
+    }
+  }
+
+  /**
+   * Whether the list element is displayed.
+   * True unless the query is a URL and adding external links is allowed.
+   */
+  protected get listIsShown(): boolean {
+    if (this.args.allowAddingExternalLinks) {
+      return !this.queryIsURL;
+    } else {
+      return true;
+    }
+  }
+
+  /**
+   * Whether to show a header above the search results (e.g., "suggestions", "results")
+   * True when there's results to show.
+   */
+  protected get listHeaderIsShown(): boolean {
+    if (this.noMatchesFound) {
+      return false;
+    }
+
+    if (this.args.allowAddingExternalLinks) {
+      return !this.queryIsURL;
+    }
+
+    return true;
+  }
+  /**
+   * Whether the query is empty.
+   * Helps determine whether the "no results" message.
+   */
+  private get queryIsEmpty(): boolean {
+    return this.query.length === 0;
+  }
+
+  /**
+   * Whether the "no results" message is hidden.
+   * False when a search error is shown, or when,
+   * if allowing external links, the query is a URL or empty.
+   */
+  protected get noResultsMessageIsHidden(): boolean {
+    if (this.args.searchErrorIsShown) {
+      return false;
+    }
+    if (this.args.allowAddingExternalLinks) {
+      return this.queryIsURL || this.queryIsEmpty;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * The action to disable keyboard navigation.
+   * Called when the search input loses focus.
+   */
+  @action protected disableKeyboardNav() {
+    this.keyboardNavIsEnabled = false;
+  }
+
+  /**
+   * The action to enable keyboard navigation.
+   * Called when the search input receives focus.
+   */
+  @action protected enableKeyboardNav() {
+    this.keyboardNavIsEnabled = true;
+  }
+
+  @action protected onItemClick(_item: any, attrs: any) {
```

Comment:
No biggie, but just noticed that this was the only action without a comment, in case it was an oversight.